contrib/fparser/fparser_ad.cc

```diff
@@ -345,11 +357,39 @@ int FunctionParserADBase<Value_t>::AutoDiff(const std::string& var_name)
 {
   this->ForceDeepCopy();
   mData = this->getParserData();
-  return ad->AutoDiff(var_name);
+  return ad->AutoDiff(LookUpVarOpcode(var_name));
 }
 
 template<typename Value_t>
-int ADImplementation<Value_t>::AutoDiff(const std::string& var_name)
+unsigned int
+FunctionParserADBase<Value_t>::LookUpVarOpcode(const std::string & var_name)
+{
+  // get c string and length of var argument
+  const unsigned len = unsigned(var_name.size());
+  const char* name = var_name.c_str();
+
+  typename FUNCTIONPARSERTYPES::NamePtrsMap<Value_t> & NamePtrs = mData->mNamePtrs;
+  typename FUNCTIONPARSERTYPES::NamePtrsMap<Value_t>::iterator vi;
+  for (vi = NamePtrs.begin(); vi != NamePtrs.end(); ++vi)
+    if (len == vi->first.nameLength && std::memcmp(name, vi->first.name, len) == 0)
```

Comment:
I'm trying to be _minimally invasive_ and try to keep all my contributions in separate derived classes to maintain a chance of upgrading to a newer fparser upstream release if it ever came out. I have contributed my core changes (bug fixes) to upstream.